Group math package functions

janekdb opened the pull request:
Group math package functions

Groups
- Mathematical Constants
- Minimum and Maximum
- Rounding
- Exponential and Logarithmic
- Trigonometric
- Angular Measurement Conversion
- Hyperbolic
- Absolute Values
- Signs
- Root Extraction
- Polar Coordindates
- Unit of Least Precision

Other changes,
- Dropped use of `double` for `Double` in all cases
- Grouped some methods in the source
- Extended notes about exception to method forwarding
- Minor method documentation enhancements

IEEERemainder is in the Rounding group since it is related to rounding and did not justify a new group.
## Review Snapshot

The Scaladoc is published here for review purposes: http://janekdb.github.io/scala/PR-5225/scala/math/index.html

janekdb committed:
Fixed a typo in Predef

janekdb commented:
I have add SI-9813 to cover the addition of the missing Java 1.6 and 1.8 methods. Happy to pick this up following agreement it should be done.

janekdb commented:
Review by anyone motivated sufficiently to do so!

som-snytt commented on src/library/scala/Predef.scala:
```diff
@@ -71,7 +71,7 @@ import scala.io.StdIn
  * @groupname assertions Assertions
  * @groupprio assertions 20
- * @groupdesc assertions These methods support program verfication and runtime correctness.
+ * @groupdesc assertions These methods support program verification and runtime correctness.
```
Are you sure it wasn't supposed to be `nerfication`?

janekdb replied:
Absolutely sure it should be `vitrification`.

janekdb commented:
Review by @martijnhoekstra . Any comments welcome.

janekdb commented:
Thanks @martijnhoekstra  !

adriaanm commented on src/library/scala/math/package.scala:
```diff
@@ -11,35 +11,98 @@ package scala
+  * @groupname root-extraction Root Extraction
+  * @groupprio root-extraction 100
+  *
+  * @groupname polar-coords Polar Coordindates
```
As in "Waiter! There's a cord in my dates!?" 😎

janekdb replied:
Fixed!

adriaanm commented on src/library/scala/math/package.scala:
```diff
@@ -11,35 +11,98 @@ package scala
+  * @groupname ulp Unit of Least Precision
+  * @groupprio ulp 120
+  *
+  * @groupname randomisation Psuedo Random Number Generation
```
`s/ue/eu/`

janekdb replied:
Fixed!

adriaanm commented:
Very nice! Caught a few typos, but otherwise looks really good!

janekdb committed:
Group math package functions

Groups

 - Mathematical Constants
 - Minimum and Maximum
 - Rounding
 - Exponential and Logarithmic
 - Trigonometric
 - Angular Measurement Conversion
 - Hyperbolic
 - Absolute Values
 - Signs
 - Root Extraction
 - Polar Coordindates
 - Unit of Least Precision

Other changes,

 - Dropped use of `double` for `Double` in all cases
 - Grouped some methods in the source
 - Extended notes about exception to method forwarding
 - Minor method documentation enhancements

IEEERemainder is in the Rounding group since it is related to rounding
and did not justify a new group.

janekdb commented:
Fixes applied, rebased and pushed.

janekdb commented:
Build failure

```
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000007a7900000, 80740352, 0) failed; error='Cannot allocate memory' (errno=12)
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 80740352 bytes for committing reserved memory.
# An error report file with more information is saved as:
# /home/jenkins/workspace/scala-2.12.x-validate-test/hs_err_pid28480.log
```

janekdb commented:
/rebuild

janekdb commented:
@adriaanm - Typos fixed, build ok. Please rereview.

som-snytt commented on src/library/scala/math/package.scala:
```diff
@@ -48,155 +111,217 @@ package object math {
+    */
   def signum(x: Int): Int       = java.lang.Integer.signum(x)
+  /** @group signum
+    * @note Forwards to [[java.lang.Long]]
```
Intended to omit note that return is widened? Or is that too banal?

som-snytt replied:
nm, I'd forgotten I'd seen it at top...

som-snytt commented:
LGTM
